File: frontend/src/types/project.ts
export type ProjectStatus =
  | "created"
  | "document_uploaded"
  | "processing"
  | "detection_completed"
  | "cad_generated"
  | "error";

export interface Project {
  id: string;
  name: string;
  description: string | null;
  status: ProjectStatus;
  document_count: number;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/project.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class ProjectCreateRequest(BaseModel):
    name: str
    description: str | None = None


class ProjectUpdateRequest(BaseModel):
    name: str | None = None
    description: str | None = None


class ProjectResponse(BaseModel):
    id: UUID
    user_id: UUID
    name: str
    description: str | None
    status: str
    document_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
